jar-submission-test/jar-submission-job/src/main/java/com/hazelcast/jet/tests/jarsubmission/pipeline/Main.java

```diff
@@ -0,0 +1,68 @@
+package com.hazelcast.jet.tests.jarsubmission.pipeline;
+
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.Util;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.test.TestSources;
+import com.hazelcast.jet.tests.jarsubmission.person.Person;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Main {
+
+    public static void main(String[] args) {
+        long index = Long.parseLong(args[0]);
+        String prefix = args[1];
+        boolean batch = Boolean.parseBoolean(args[2]);
+
+        Pipeline pipeline = batch ? batchPipeline(index, prefix) : streamPipeline(index, prefix);
+        HazelcastInstance hz = Hazelcast.bootstrappedInstance();
+        hz.getJet().newJob(pipeline);
+    }
+
+    private static Pipeline batchPipeline(long index, String prefix) {
+        List<String> source = new ArrayList<>();
+        source.add(prefix + index);
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(TestSources.items(source))
+                .rebalance()
+                .map(t -> new Person(t))
+                .rebalance()
+                .map(t -> new AnotherPerson(t.getName()))
+                .rebalance()
+                .map(t -> {
+                    if (index % 10 == 0) {
+                        throw new RuntimeException();
+                    }
+                    return t;
+                })
+                .rebalance()
+                .map(t -> Util.entry(prefix, t.getName()))
+                .writeTo(Sinks.map(prefix));
+        return pipeline;
+    }
+
+    private static Pipeline streamPipeline(long index, String prefix) {
+        String item = prefix + index;
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(TestSources.itemStream(10, (t, u) -> item))
```

Comment:
I think you may only use TestSources.itemStream(10) to create 10 person objects per second. Since all Person objects use the same String item for the constructor. I don't think you need the generator function

Reply:
I want to generate same items (since I'm basically interested in the first of them and don't want to store more items to map then is needed).

I can do in the way as it is now (by generator function) or I can move the logic to the first mapping stage where I can use `item` for creating `Person` like `.map(t -> new Person(item))`.

If I'm think about it the second approach can be actually more readable => will change it to `.map(t -> new Person(item))` and remove generator function.